source/features/previous-version.tsx

```diff
@@ -0,0 +1,117 @@
+import * as pageDetect from 'github-url-detection';
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+
+import features from '../feature-manager';
+import * as api from '../github-helpers/api';
+import observe from '../helpers/selector-observer';
+
+import GitHubURL from '../github-helpers/github-url';
+
+const getHistoryOids = cache.function('previous-version', async (githubUrl: GitHubURL): Promise<string[] | false> => {
+	const {resource: {history}} = await api.v4(`
+		resource(url: "/${githubUrl.user}/${githubUrl.repository}/commit/${githubUrl.branch}") {
+			... on Commit {
+				history(path: "${githubUrl.filePath}") {
```

Comment:
Let's limit this to 4. I don’t think it's particularly useful to go to a random "15 commits ago"

I'm not sure we even need more than "previous version." Can we think of a reason to just jump to "two unnamed commits ago"?

Reply:
I can, but not more than 4.

Reply:
Yeah, not providing like Jetbrains's Git history. I think no need to go back over 4 commits.
If we show "commit message", then showing over 4 commits would be usefull

Reply:
> I can

Go on? 😅 what's the reason to go blindly back two commits?

Reply:
On code that I wrote I usually remember what I did.

But your right, on a random repo more than 2 is sort of blind.